test/context/yaml/context.test.js

```diff
@@ -116,7 +116,80 @@ describe('#YAML context validation', () => {
       ],
       connections: [],
       databases: [],
-      emailProvider: {},
+      emailProvider: {
+        enabled: true,
+        name: 'smtp',
+        credentials: {
+          smtp_host: '##SMTP_HOSTNAME##',
+          smtp_pass: '##SMTP_PASS##',
+          smtp_port: '##SMTP_PORT##',
+          smtp_user: '##SMTP_USER##'
+        }
+      },
+      emailTemplates: [
+        { body: './emailTemplates/blocked_account.html', enabled: true, template: 'blocked_account' },
+        { body: './emailTemplates/change_password.html', enabled: true, template: 'change_password' },
+        { body: './emailTemplates/enrollment_email.html', enabled: true, template: 'enrollment_email' },
+        { body: './emailTemplates/mfa_oob_code.html', enabled: true, template: 'mfa_oob_code' },
+        { body: './emailTemplates/password_reset.html', enabled: true, template: 'password_reset' },
+        { body: './emailTemplates/reset_email.html', enabled: true, template: 'reset_email' },
+        { body: './emailTemplates/stolen_credentials.html', enabled: true, template: 'stolen_credentials' },
+        { body: './emailTemplates/verify_email.html', enabled: true, template: 'verify_email' },
+        { body: './emailTemplates/welcome_email.html', enabled: true, template: 'welcome_email' }
+      ],
+      pages: [
+        { enabled: true, html: './pages/login.html', name: 'login' }
+      ],
+      guardianFactors: [],
+      guardianFactorProviders: [],
+      guardianFactorTemplates: [],
+      prompts: {},
+      resourceServers: [],
+      rules: [],
+      hooks: [],
+      rulesConfigs: [],
+      roles: [
+        {
+          name: 'App Admin',
+          description: 'Admin of app',
+          permissions: [
+            {
+              permission_name: 'create:data', resource_server_identifier: 'urn:ref'
+            }
+          ]
+        }
+      ],
+      tenant: {
+        default_directory: 'users',
+        friendly_name: 'Test'
+      }
+    });
+  });
+
+  it('should dump tenant.yaml without defaults', async () => {
+    const dir = path.resolve(testDataDir, 'yaml', 'dump');
+    cleanThenMkdir(dir);
+    const tenantFile = path.join(dir, 'tenant.yml');
+    const config = { AUTH0_INPUT_FILE: tenantFile, AUTH0_EXCLUDED_DEFAULTS: [ 'emailProvider' ] };
+    const context = new Context(config, mockMgmtClient());
+    await context.dump();
+    const yaml = jsYaml.safeLoad(fs.readFileSync(tenantFile));
+    expect(yaml).to.deep.equal({
+      branding: {},
+      clientGrants: [],
+      clients: [
+        {
+          custom_login_page: './Global Client_custom_login_page.html',
+          custom_login_page_on: true,
+          name: 'Global Client'
+        }
+      ],
+      connections: [],
+      databases: [],
+      emailProvider: {
+        enabled: true,
+        name: 'smtp'
+      },
```

Comment:
So `AUTH0_EXCLUDED_DEFAULTS: [ 'emailProvider' ]` is excluding just `emailProvider.credentials` but not necessarily the whole object?

Reply:
Since we are not getting `emailProvider.credentials` from management API, the behavior before `AUTH0_EXCLUDED_DEFAULTS` was to provide some defaults. This flag just disables the filling of these defaults.